kart/checkout.py

```diff
@@ -21,6 +22,13 @@
 )
 
 
+def checkout_completer(ctx, param, incomplete):
+    current_working_directory = os.getcwd()
+    repository_path = pygit2.discover_repository(current_working_directory)
+    repo = pygit2.Repository(repository_path)
+    return [b for b in repo.listall_branches() if b.startswith(incomplete)]
```

Comment:
I'd suggest matching refs too using itertools.chain
```suggestion
    all_refs = chain(repo.listall_branches(), repo.references)
    return [b for b in all_refs if b.startswith(incomplete)]
```